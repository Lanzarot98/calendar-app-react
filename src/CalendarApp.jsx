import { BrowserRouter, HashRouter } from "react-router-dom"
import { AppRouter } from "./router"


export const CalendarApp = () => {
  return (
    <BrowserRouter>
      <AppRouter />
    </BrowserRouter>
  )
}
