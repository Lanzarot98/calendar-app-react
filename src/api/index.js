


// coloco el as para que se conozca la exportaci√≥n por defecto, porque se hizo un export default
export * as calendarApi from './calendarApi';
