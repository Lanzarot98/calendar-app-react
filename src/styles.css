body {
    background-color: rgb(255, 255, 255);
    color: black;
}

.rbc-ellipsis, .rbc-show-more, .rbc-row-segment .rbc-event-content, .rbc-event-label {
    white-space: normal;
}