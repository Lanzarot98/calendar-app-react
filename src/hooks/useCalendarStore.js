import { useDispatch, useSelector } from "react-redux"
import { calendarApi } from "../api";
import { convertEventsToDateEvents } from "../helpers";
import { onAddNewEvent, onDeleteEvent, onSetActiveEvent, onUpdateEvent } from "../store";


export const useCalendarStore = () => {
  
    const dispatch = useDispatch();
    const { events, activeEvent } = useSelector( state => state.calendar );
    const { user } = useSelector( state => state.auth );

    const setActiveEvent = ( calendarEvent ) => {
        dispatch( onSetActiveEvent( calendarEvent ) )
    }

    const startSavingEvent = async( calendarEvent ) => {
        // TODO: llegar al backend

        // Todo Update
        if( calendarEvent._id ) {
            // actualizando
            dispatch( onUpdateEvent({ ...calendarEvent }) );
        } else {
            // creando
            const { data } = await calendarApi.post('/events', calendarEvent);
            // console.log({data});

            dispatch( onAddNewEvent({ ...calendarEvent, id: data.event.id, user }) )
        }

    }

    const startDeletingEvent = () => {
        // Todo: llevar al backend
        dispatch( onDeleteEvent() )
    }

    const startLoadingEvents = async() => {
        try {
            
            const { data } = await calendarApi.get('/events');
            const events = convertEventsToDateEvents( data.events );
            console.log({data});
            console.log(events);


        } catch (error) {
            console.log('Error Cargando Eventos');
            console.log(error)
        }
    }

    return {
        // propiedades:
        activeEvent,
        events,
        hasEventSelected: !!activeEvent?._id, //si es null regresa falso, si tiene un objeto regresa true

        // m√©todos:
        setActiveEvent,
        startSavingEvent,
        startDeletingEvent,
        startLoadingEvents,
    }
}
