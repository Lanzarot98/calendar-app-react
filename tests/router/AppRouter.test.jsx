import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { CalendarPage } from "../../src/calendar";
import { useAuthStore } from "../../src/hooks/useAuthStore";
import { AppRouter } from "../../src/router/AppRouter";

jest.mock('../../src/hooks/useAuthStore');

jest.mock('../../src/calendar', () => ({
    CalendarPage: () => <h1>CalendarPage</h1>
}))


describe('Pruebas en <AppRouter />', () => { 
    
    const mockCheckAuthToken = jest.fn();

    beforeEach(() => jest.clearAllMocks());

    test('debe de mostrar la pantalla de carga y llamar checkAuthToken', () => { 
        
        useAuthStore.mockReturnValue({
            // errorMessage: undefined,
            status: 'checking',
             checkAuthToken: mockCheckAuthToken
            // user: {}
        });

        render(<AppRouter/>);
        // screen.debug();
        expect( screen.getByText('Cargando...') ).toBeTruthy();
        expect( mockCheckAuthToken ).toHaveBeenCalled();

    });

    test('debe de mostrar el login en caso de nos estar autenticado', () => {   

        useAuthStore.mockReturnValue({
            status: 'not-authenticated',
            checkAuthToken: mockCheckAuthToken
        });
        const { container } = render(
            <MemoryRouter initialEntries={['/auth/algo']} >
                <AppRouter/>
            </MemoryRouter>
        );
        // screen.debug();
        expect( screen.getByText('Login') ).toBeTruthy();
        expect( container ).toMatchSnapshot();

    });
    
    test('debe de mostrar el calendario si estamos autenticados', () => {   

        useAuthStore.mockReturnValue({
            status: 'authenticated',
            checkAuthToken: mockCheckAuthToken
        });
        render(
            <MemoryRouter >
                <AppRouter/>
            </MemoryRouter>
        );
        // screen.debug();
        expect( screen.getByText('CalendarPage') ).toBeTruthy();
        

    });

});