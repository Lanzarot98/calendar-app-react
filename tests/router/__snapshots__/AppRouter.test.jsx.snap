// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el login en caso de nos estar autenticado 1`] = `
<div>
  <div
    class="container login-container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 login-form-1"
      >
        <h3>
          Login to 
          <strong>
            Calendar App
          </strong>
          ðŸ“†
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="loginEmail"
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="loginPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-2 text-center"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        class="col-md-6 login-form-2"
      >
        <h3>
          Sign up
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerName"
              placeholder="Full name"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerEmail"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerPassword2"
              placeholder="Confirm password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-2 text-center"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Sign up"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
